File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/ClickCommand_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "ClickCommand_ViewModelTestEnvironment.h"
#include "ClickCommand_ViewModel.h"
#include "ClickCommand_ViewModelTestEnvironmentImpl.h"

namespace commands
{
  class ClickCommand_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<ClickCommand_ViewModel> sut;
    std::shared_ptr<ClickCommand_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    // When Helper Definitions
    virtual void when_click_Ok();
  protected:
    void SetUp() override;
  };
  void ClickCommand_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<ClickCommand_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  /// Scenario: Click Button
  ///   given:
  ///    when: click Ok
  ///    then:
  TEST_F(ClickCommand_ViewTests, Click_Button_given_when_click_Ok_then_)
  {
    this->BuildSut();
    this->when_click_Ok();
  }
  void ClickCommand_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void ClickCommand_ViewTests::when_click_Ok()
  {
    this->sut->okButtonClicked();
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/ClickCommand_ViewTests.cs
/// <filename>
///     ClickCommand_ViewTests.cs
/// </filename>
using commands;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class ClickCommand_ViewTests
{
    private commands.ClickCommand_ViewModel sut;

    private commands.ClickCommand_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new commands.ClickCommand_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void Click_Button_given_when_click_Ok_then_()
    {
        this.BuildSut();
        this.when_click_Ok();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    public virtual void when_click_Ok()
    {
        this.sut.okButtonClicked();
    }
}
